packages/storage/amplify_storage_s3/example/integration_test/upload_file_test.dart

```diff
@@ -278,58 +278,110 @@ void main() {
+                );
+              }
+            },
           );
-          await Future<void>.delayed(const Duration(seconds: 15));
-        });
 
-        testWidgets('can resume', (_) async {
-          final fileId = uuid();
-          final path = 'public/upload-file-resume-$fileId';
-          final filePath = await createFile(path: fileId, content: content);
-          addTearDownPath(StoragePath.fromString(path));
-          final operation = Amplify.Storage.uploadFile(
-            localFile: AWSFile.fromPath(filePath),
-            path: StoragePath.fromString(path),
+          testWidgets(
+            'can resume (file size: $fileSize mb)',
+            (_) async {
+              final fileId = uuid();
+              final path = 'public/upload-file-resume-$fileId';
+              final filePath = await createFile(path: fileId, content: content);
+              final state = StreamController<StorageTransferState>();
+              addTearDownPath(StoragePath.fromString(path));
+              final operation = Amplify.Storage.uploadFile(
+                localFile: AWSFile.fromPath(filePath),
+                path: StoragePath.fromString(path),
+                onProgress: (progress) {
+                  state.sink.add(progress.state);
+                },
+              );
+              await operation.pause();
+              await operation.resume();
+              final nextProgressState = await state.stream.first;
+              expect(nextProgressState, StorageTransferState.inProgress);
+              final result = await operation.result;
+              expect(result.uploadedItem.path, path);
+              final downloadResult = await Amplify.Storage.downloadData(
+                path: StoragePath.fromString(path),
+              ).result;
+              expect(downloadResult.bytes, content.codeUnits);
+              await state.close();
+            },
           );
-          await operation.pause();
-          await operation.resume();
-          final result = await operation.result;
-          expect(result.uploadedItem.path, path);
-        });
 
-        testWidgets('can cancel', (_) async {
-          final fileId = uuid();
-          final path = 'public/upload-file-cancel-$fileId';
-          final filePath = await createFile(path: fileId, content: content);
-          addTearDownPath(StoragePath.fromString(path));
-          final operation = Amplify.Storage.uploadFile(
-            localFile: AWSFile.fromPath(filePath),
-            path: StoragePath.fromString(path),
-          );
-          final expectException = expectLater(
-            () => operation.result,
-            throwsA(isA<StorageOperationCanceledException>()),
-          );
-          await operation.cancel();
-          await expectException;
-        });
+          testWidgets('can cancel (file size: $fileSize mb)', (_) async {
+            final fileId = uuid();
+            final path = 'public/upload-file-cancel-$fileId';
```

Comment:
should this also be checking if the file size is 5mb like pause test since small file uploads can't be canceled??

Reply:
Cancel is supported for both single part and multi part uploads. Only resume/pause are not supported for single part uploads